### PR TITLE
fix(autoware_image_projection_based_fusion): detected object roi box projection fix

### PR DESCRIPTION
**Pull request by technolojin**
## Description

1. eliminate misuse of `std::numeric_limits<double>::min()`
2. fix roi range up to the image edges

## Related links

**Parent Issue:**

- Link [TIER IV INTERNAL](https://star4.slack.com/archives/CTEJP8L4T/p1732608837719329)

## How was this PR tested?

Screenshot from 2024-12-02 15-09-33

Screenshot from 2024-12-02 15-17-24

Objects at the edges are fused without issue.

## Notes for reviewers
It will be merged after https://github.com/autowarefoundation/autoware.universe/pull/9505 is merged and the build issue is resolved.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: perception/autoware_image_projection_based_fusion/src/roi_detected_object_fusion/node.cpp [hunk 1]
```diff
@@ -172,18 +171,16 @@ RoiDetectedObjectFusionNode::generateDetectedObjectRoIs(
       continue;
     }
 
-    min_x = std::max(min_x, 0.0);
-    min_y = std::max(min_y, 0.0);
-    max_x = std::min(max_x, image_width - 1);
-    max_y = std::min(max_y, image_height - 1);
+    const uint32_t idx_min_x = std::floor(std::max(min_x, 0.0));
+    const uint32_t idx_min_y = std::floor(std::max(min_y, 0.0));
+    const uint32_t idx_max_x = std::ceil(std::min(max_x, image_width));
```
[the remark this answers is not in the document]
Comment (technolojin) [reply perception/autoware_image_projection_based_fusion/src/roi_detected_object_fusion/node.cpp:176]: The roi boundary is the coordinate of image in pixel. 
If we think 1 by 1 pixel image and 1 by 1 ROI, min is (0,0) and max is (1,1).
Therefore, the possible maximum value should be the `image_width`, not `image_width - 1`

[the remark this answers is not in the document]
Comment (technolojin) [reply perception/autoware_image_projection_based_fusion/src/roi_detected_object_fusion/node.cpp:176]: If the following nodes expect the maximum value to `image_width - 1`, we may need to align relevant nodes.

[the remark this answers is not in the document]
Comment (yukkysaito) [reply perception/autoware_image_projection_based_fusion/src/roi_detected_object_fusion/node.cpp:176]: Understood. :+1: 
Have you conducted a simple test for verification?

[the remark this answers is not in the document]
Comment (technolojin) [reply perception/autoware_image_projection_based_fusion/src/roi_detected_object_fusion/node.cpp:176]: @yukkysaito 
Tests on objects at image edges are done and the results are visualized as the description.

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**